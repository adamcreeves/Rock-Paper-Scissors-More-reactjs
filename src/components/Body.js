import React, { useState } from "react";
import Title from "./subcomponents/Title";
import GameBody from "./subcomponents/GameBody";
import RegisterPlayers from "./subcomponents/RegisterPlayers";
import {
  str_001,
  str_002,
  str_003,
  str_073,
  str_074,
} from "../resources/strings";
import { c_001, c_002, c_003 } from "../resources/classNames";
import Cookies from "universal-cookie";
import "../styles/Main.css";

function Body() {
  const cookies = new Cookies();
  const storedPlayer1 = cookies.get(str_073) || str_001;
  const storedPlayer2 = cookies.get(str_074) || str_001;
  const [player1, setPlayer1] = useState(storedPlayer1);
  const [player2, setPlayer2] = useState(storedPlayer2);
  const resetGame = () => {
    setPlayer1(str_001);
    setPlayer2(str_001);
    cookies.remove(str_073);
    cookies.remove(str_074);
  };
  return (
    <div className={c_001} data-testid="body-component">
      <Title str={str_002} classNm={c_002} />
      <Title str={str_003} classNm={c_003} />
      {!player1 ? (
        <RegisterPlayers setPlayer1={setPlayer1} setPlayer2={setPlayer2} />
      ) : (
        <GameBody
          player1={player1}
          player2={player2}
          resetPlayers={resetGame}
        />
      )}
    </div>
  );
}

export default Body;
