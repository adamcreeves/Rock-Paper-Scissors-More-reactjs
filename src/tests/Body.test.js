import React from "react";
import { render } from "@testing-library/react";
import Body from "../components/Body";

describe("Test Body", () => {
  it("should match the snapshot", () => {
    const { asFragment } = render(<Body />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("should render Body and not be null", () => {
    const { asFragment } = render(<Body />);
    expect(asFragment()).not.toBeNull();
  });
});
