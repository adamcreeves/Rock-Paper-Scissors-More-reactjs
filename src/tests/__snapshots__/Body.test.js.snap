// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Body should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="body"
    data-testid="body-component"
  >
    <div
      class="title"
      data-testid="title-component"
    >
      Welcome to Adam's
    </div>
    <div
      class="subtitle"
      data-testid="title-component"
    >
      Rock Paper Scissors Game
    </div>
    <form
      class="registerPlayers"
    >
      <input
        class="registerPlayers__playerName"
        placeholder="Enter Player 1's Name"
        required=""
        type="text"
        value=""
      />
      <input
        class="registerPlayers__playerName"
        placeholder="Enter Player 2's Name"
        required=""
        type="text"
        value=""
      />
      <input
        class="gameBody__gameButton"
        type="submit"
        value="Start Game"
      />
    </form>
  </div>
</DocumentFragment>
`;
